sympy/physics/control/lti.py

```diff
@@ -913,75 +1057,125 @@ def doit(self, **kwargs):
             if res is None:
                 res = arg
             else:
-                num_ = res.num * arg.den + res.den * arg.num
-                den_ = res.den * arg.den
-                res = TransferFunction(num_, den_, self.var)
+                if self.is_SISO:
+                    if res.den == arg.den:
+                        num_, den_ = res.num + arg.num, res.den
+                    else:
+                        num_, den_ = res.num * arg.den + res.den * arg.num, res.den * arg.den
+                    res = TransferFunction(num_, den_, self.var)
+                else:
+                    if self.num_inputs == 1:
+                        a = [None] * self.num_outputs
+                        for x in range(self.num_outputs):
+                            a[x] = res.args[0][x] + arg.args[0][x]
+                    else:
+                        a = [[None] * self.num_inputs for _ in range(self.num_outputs)]
+                        for row in range(self.num_outputs):
+                            for col in range(self.num_inputs):
+                                a[row][col] = res.args[0][row][col] + arg.args[0][row][col]
+                    res = TransferFunctionMatrix(a, self.shape, self.var)
```

Comment:
I think this would be clearer with functools reduce:

```python
if self.is_SISO:
    def mul(res, arg):
        arg = arg.doit()
        if res.den == arg.den:
            num_, den_ = res.num + arg.num, res.den
        else:
            num_, den_ = res.num * arg.den + res.den * arg.num, res.den * arg.den
        return TransferFunction(num_, den_, self.var)
else:
    def mul(res, arg):
        arg = arg.doit()
        if self.num_inputs == 1:
            a = [None] * self.num_outputs
            for x in range(self.num_outputs):
                a[x] = res.args[0][x] + arg.args[0][x]
        else:
            a = [[None] * self.num_inputs for _ in range(self.num_outputs)]
            for row in range(self.num_outputs):
                for col in range(self.num_inputs):
                    a[row][col] = res.args[0][row][col] + arg.args[0][row][col]
        return TransferFunctionMatrix(a, self.shape, self.var)

return functools.reduce(self.args, mul)
```

Reply:
using this diff, I get:
> TypeError: reduce() arg 2 must support iteration

Reply:
I got the arguments in the wrong order, like I always do